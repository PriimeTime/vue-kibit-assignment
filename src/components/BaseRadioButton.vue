<script setup lang="ts">
const props = defineProps<{
  modelValue: string;
  value: string;
  id: string;
  name: string;
}>();

const emit = defineEmits(["update:modelValue"]);
</script>

<template>
  <label class="flex items-center space-x-2">
    <input
      type="radio"
      :id="props.id"
      :name="props.name"
      :value="value"
      :checked="modelValue === value"
      @change="$emit('update:modelValue', props.value)"
    />
    <span class="text-teal-900">
      <slot />
    </span>
  </label>
</template>
