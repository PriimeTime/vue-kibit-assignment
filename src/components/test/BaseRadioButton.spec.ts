import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import BaseRadioButton from "@/components/BaseRadioButton.vue";

describe("RadioButton", () => {
  it("renders without crashing", () => {
    const wrapper = mount(BaseRadioButton, {
      props: {
        id: "option1",
        name: "group1",
        value: "value1",
        modelValue: "",
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("binds the 'checked' attribute based on modelValue", () => {
    const wrapper = mount(BaseRadioButton, {
      props: {
        id: "option1",
        name: "group1",
        value: "value1",
        modelValue: "value1",
      },
    });
    const input = wrapper.find("input");
    expect(input.element.checked).toBe(true);
  });

  it("emits 'update:modelValue' when the value changes", async () => {
    const wrapper = mount(BaseRadioButton, {
      props: {
        id: "option1",
        name: "group1",
        value: "value1",
        modelValue: "",
      },
    });
    const input = wrapper.find("input");

    await input.setValue();
    expect(wrapper.emitted("update:modelValue")).toBeTruthy();
    expect(wrapper.emitted("update:modelValue")![0]).toEqual(["value1"]);
  });

  it("matches snapshot", () => {
    const wrapper = mount(BaseRadioButton, {
      props: {
        id: "option1",
        name: "group1",
        value: "value1",
        modelValue: "value2",
      },
      slots: {
        default: "Option 1",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
