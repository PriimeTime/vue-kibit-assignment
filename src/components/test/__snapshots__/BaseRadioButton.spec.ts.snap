// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioButton > matches snapshot 1`] = `"<label class="flex items-center space-x-2"><input type="radio" id="option1" name="group1" value="value1"><span class="text-teal-900">Option 1</span></label>"`;
